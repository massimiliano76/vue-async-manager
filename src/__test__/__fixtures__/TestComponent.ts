import { CreateElement } from 'vue'

export default {
  name: 'TestComponent',
  render(h: CreateElement) {
    return h('div', { class: { 'test-component': true } }, 'test component')
  }
}
