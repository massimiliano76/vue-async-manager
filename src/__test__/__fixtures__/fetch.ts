import { Component } from 'vue'

export default () => {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve({
        code: 0,
        msg: 'success',
        data: {}
      })
    }, 1000)
  })
}

export function dynamicImport(Comp: Component): Promise<Component> {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve(Comp)
    }, 1000)
  })
}
