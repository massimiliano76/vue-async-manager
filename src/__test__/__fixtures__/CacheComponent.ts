import { CreateElement } from 'vue'
import createResource from '../../createResource'
import fetch from './fetch'

export default {
  name: 'CacheComponent',
  created() {
    this.$rm = createResource(() => fetch())
    // This.promiser is used in test cases, waiting for resource requests to complete
    this.promiser = this.$rm.read()
  },
  render(h: CreateElement) {
    return h(
      'div',
      { class: { 'cache-component': true } },
      JSON.stringify(this.$rm.$result)
    )
  }
}
