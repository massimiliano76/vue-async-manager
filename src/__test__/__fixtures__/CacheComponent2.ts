import { CreateElement } from 'vue'
import createResource from '../../createResource'
import fetch from './fetch'

export default {
  name: 'CacheComponent2',
  data() {
    return {
      res: {}
    }
  },
  async created() {
    this.resource = createResource(() => fetch())
    // This.promiser is used in test cases, waiting for resource requests to complete
    this.promiser = this.resource.read()
    await this.promiser
    this.res = this.resource.$res.$$result
  },
  render(h: CreateElement) {
    return h(
      'div',
      { class: { 'cache-component-2': true } },
      JSON.stringify(this.res)
    )
  }
}
