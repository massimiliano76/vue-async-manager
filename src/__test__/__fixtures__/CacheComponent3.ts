import { CreateElement } from 'vue'
import createResource from '../../createResource'
import fetch from './fetch'

const resource = createResource(() => fetch())

export default {
  name: 'CacheComponent3',
  data() {
    return {
      res: {}
    }
  },
  async created() {
    const res = resource.read()
    // This.promiser is used in test cases, waiting for resource requests to complete
    this.promiser = res.$$waiter
    await res.$$waiter
    this.res = res.$$result
  },
  render(h: CreateElement) {
    console.log('CacheComponent render')
    return h(
      'div',
      { class: { 'cache-component-3': true } },
      JSON.stringify(this.res)
    )
  }
}
