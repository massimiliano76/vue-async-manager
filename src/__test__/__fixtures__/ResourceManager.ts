import { CreateElement } from 'vue'
import createResource from '../../createResource'
import fetch from './fetch'

export default {
  name: 'ResourceManager',
  created() {
    this.$rm = createResource((mockData: any) => fetch(mockData))
    this.$rm2 = this.$rm.fork()

    this.promiser1 = this.$rm.read({ name: 'foo' })
    this.promiser2 = this.$rm2.read({ name: 'bar' })
  },
  render(h: CreateElement) {
    return h('div')
  }
}
