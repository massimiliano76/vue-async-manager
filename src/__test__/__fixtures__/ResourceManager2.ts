import { CreateElement } from 'vue'
import createResource from '../../createResource'
import fetch from './fetch'

export default {
  name: 'ResourceManager',
  created() {
    this.$rm = createResource((mockData: any) => fetch(mockData), {
      prevent: true
    })

    this.promiser1 = this.$rm.read({ name: 'foo' })
    this.promiser2 = this.$rm.read({ name: 'bar' })

    this.$rm2 = createResource((mockData: any) => fetch(mockData))

    this.promiser3 = this.$rm2.read({ name: 'foo' })
    this.promiser4 = this.$rm2.read({ name: 'bar' })
  },
  render(h: CreateElement) {
    return h('div')
  }
}
