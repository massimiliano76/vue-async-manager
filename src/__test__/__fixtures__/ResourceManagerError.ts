import { CreateElement } from 'vue'
import createResource from '../../createResource'
import fetch from './fetch'

export default {
  name: 'ResourceManager',
  created() {
    this.$rm = createResource((mockData: any) =>
      fetch(mockData, true /* throw error */)
    )

    this.promiser = this.$rm.read({ name: 'foo' })
  },
  render(h: CreateElement) {
    return h('div')
  }
}
