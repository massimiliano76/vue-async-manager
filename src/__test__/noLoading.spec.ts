import Vue from 'vue'
import installer from '../index'
import lazy from '../lazy'
import TestComp from './__fixtures__/TestComponent'
import CacheComponent3 from './__fixtures__/CacheComponent3'
import { dynamicImport } from './__fixtures__/fetch'

Vue.config.devtools = false
Vue.config.productionTip = false

describe('Loading:', () => {
  Vue.use(installer)

  test('Nested Suspense components', async () => {
    const TestComponent = lazy(() => dynamicImport(TestComp))

    const ins = new Vue({
      name: 'TreeComponent',
      components: {
        TestComponent,
        CacheComponent3
      },
      render(h) {
        return h('Suspense', { props: { delay: 10000 } }, [
          h('div', { class: { loading: true }, slot: 'fallback' }, 'loading'),
          h('TestComponent'),
          h('Suspense', [h('CacheComponent3')])
        ])
      }
    })

    ins.$mount()

    expect(ins.$el.outerHTML).toMatchSnapshot()

    await (ins.$children[0] as Vue).promiser
    await (ins.$children[0].$children[1] as Vue).promiser
    await (ins.$children[0].$children[1].$children[0] as Vue).promiser

    expect(ins.$el.outerHTML).toMatchSnapshot()
  })
})
