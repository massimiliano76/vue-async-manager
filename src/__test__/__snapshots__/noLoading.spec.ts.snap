// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading: Nested Suspense components 1`] = `"<div class=\\"vue-suspense-wrapper\\"><!----><div class=\\"vue-suspense-wrapper\\"><!----><!----></div><!----></div>"`;

exports[`Loading: Nested Suspense components 2`] = `"<div class=\\"vue-suspense-wrapper\\"><div class=\\"test-component\\">test component</div><div class=\\"vue-suspense-wrapper\\"><div class=\\"cache-component-3\\">{\\"code\\":0,\\"msg\\":\\"success\\",\\"data\\":{}}</div></div></div>"`;
